pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,17 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
```

Comment:
Ok, I'm fine with that for now.  We can always refactor later if we circle back, this will be labeled experimental till then anyway.

Add a line to the Godoc for this strategy that this strategy is experimental.